File: backend/burning-okr/burning-okr-app/src/main/java/org/burningokr/dto/structure/CompanyDto.java
package org.burningokr.dto.structure;

import java.util.ArrayList;
import java.util.Collection;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
public class CompanyDto extends StructureDto {

  private Long cycleId;

  private Collection<Long> subStructureIds = new ArrayList<>();

  private Long historyId;
}

File: frontend/src/app/shared/model/api/structure/company.dto.ts
import { CycleId, StructureId } from '../../id-types';
import { StructureDto } from './structure.dto';

export interface CompanyDto extends StructureDto {
  cycleId: CycleId;
  subStructureIds?: StructureId[];
  corporateObjectiveIds?: number[];
}
